components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/SynapseAnalyticsDataProvider.java

```diff
@@ -186,7 +185,11 @@ public Latencies getLatencies() {
     @Override
     public MetaInfo getMetaInfo() {
         MetaInfo metaInfo = new MetaInfo();
-        metaInfo.setCorrelationId(UUID.randomUUID().toString());
+        Object correlationId  = ((Axis2MessageContext) messageContext).getAxis2MessageContext()
+                .getProperty(Constants.CORRELATION_ID);
```

Comment:
change the correlation_id constant to use from mi as https://github.com/wso2/carbon-apimgt/pull/10004/files

Reply:
Changed